modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer.js

```diff
@@ -29,8 +29,16 @@ const DEFAULT_GAMMA = 0.2;
 const DEFAULT_BUFFER = 192.0 / 256;
 
 const defaultProps = {
-  getShiftInQueue: {type: 'accessor', value: x => x.shift || 0},
+  // each queue can have one or multiple line(s)
+  // each line can have one or multiple object(s)
+  // length of the line where the given object is
+  getLineLength: {type: 'accessor', value: x => x.lineLength || 0},
+  // offset from the left, top position of the queue
+  getOffsets: {type: 'accessor', value: x => x.shift || 0},
+  // length of the queue
   getLengthOfQueue: {type: 'accessor', value: x => x.len || 1},
+  // height of the queue
+  getHeightOfQueue: {type: 'accessor', value: x => x.height || 1},
```

Comment:
Since `getLengthOfQueue` and `getHeightOfQueue` are always called together, just merge them into one accessor. Prop comparison has a cost.

"Queue" was a really bad name. More intuitive terms here may be
- character
- column
- row
- paragraph